arbnode/transaction_streamer.go

```diff
@@ -1509,16 +1517,23 @@ func (s *TransactionStreamer) checkSubmittedTransactionForFinality(ctx context.C
 			resubmittedTxn, err := s.resubmitTransactionIfPastDelay(ctx, submittedTx)
 			if err != nil {
 				log.Error("failed to resubmit transaction", "err", err)
-				continue
 			}
 			if resubmittedTxn != nil {
 				newSubmittedTxns = append(newSubmittedTxns, *resubmittedTxn)
 			} else {
 				newSubmittedTxns = append(newSubmittedTxns, submittedTx)
 			}
-			continue
+			hasInterrupted = true
+		}
+		log.Info("transaction checked", "hash", hash, "data", data)
+
+		if !hasInterrupted {
+			dataArray = append(dataArray, data)
 		}
+	}
 
+	for i, data := range dataArray {
+		submittedTx := submittedTxns[i]
```

Comment:
I think this is wrong, We need to use the index of the submittedTxns array relative to the index of the data array.
This needs to be changed to store the submittedtxn array index when we write to the data array.